modules/scalacheck/shared/src/main/scala/eu/timepit/refined/scalacheck/numeric.scala

```diff
@@ -111,3 +111,15 @@ trait NumericInstances {
   ): Arbitrary[F[T, P]] =
     arbitraryRefType(Gen.chooseNum(min, max))
 }
+
+trait NumericInstancesBinCompat1 {
+  implicit def floatNotNaNArbitrary[F[_, _]: RefType](
+      implicit arb: Arbitrary[Float]
+  ): Arbitrary[F[Float, NotNaN]] =
+    arbitraryRefType(arb.arbitrary.suchThat(x => !x.isNaN))
```

Comment:
Minor nitpick: Using `suchThat` could lead in theory to "discarded too many values" errors. I don't think this will ever be a problem in this case but we could eliminate this possibility altogether by mapping `NaN` values to `0.0`.
```suggestion
    arbitraryRefType(arb.arbitrary.map(x => if (x.isNaN) 0.0F else x))
```